src/app/Shared/Services/Api.service.tsx

```diff
@@ -411,3 +414,16 @@ export interface EventTemplate {
   provider: string;
   type: 'CUSTOM' | 'TARGET';
 }
+
+export interface RecordingOptions {
+  toDisk: boolean;
+  maxSize: number;
+  maxAge: number;
+}
+
+export interface RecordingAttributes {
+  name: string;
+  events: string;
+  duration?: number;
+  options: RecordingOptions;
```

Comment:
If `duration` is marked as optional then I think it also makes sense to mark `options` as optional. And, within the options, each of its attributes should probably be optional as well.

Reply:
I can mark them optional, however there will always be a value for them because unlike duration, there is no scenario where the recordingOptions isn't sent.

Reply:
That may be the case *right now*, but you can't make that claim as an incontrovertible fact for all potential future modifications to the code...

The type definition should model not only how the type is currently used by the implementation, but model the general specifications and requirements. The options are not required to be sent when performing requests to the backend, so the type should mark the options as, well, optional.